src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -220,26 +220,36 @@ static ImmutableArray<IMethodSymbol> GetOperatingSystemGuardMethods(IMethodSymbo
             });
         }
 
-        private static bool HasInterproceduralResult(IOperation platformSpecificOperation, SmallDictionary<string, PlatformAttributes> attributes,
+        private static bool HasGuardedInterproceduralResult(IOperation platformSpecificOperation, SmallDictionary<string, PlatformAttributes> attributes,
             DataFlowAnalysisResult<GlobalFlowStateBlockAnalysisResult, GlobalFlowStateAnalysisValueSet> analysisResult)
         {
-            if (platformSpecificOperation.IsWithinLambdaOrLocalFunction())
+            if (!platformSpecificOperation.IsWithinLambdaOrLocalFunction())
             {
-                var results = analysisResult.TryGetInterproceduralResults();
-                if (results != null)
+                return false;
+            }
+
+            var results = analysisResult.TryGetInterproceduralResults();
+            if (!results.Any())
+            {
+                // Do not flag code inside unused lambda/local functions.
+                return true;
```

Comment:
Here we are suppressing warning for "unused lambda/local" methods, i am OK with this approach but @jeffhandley @terrajobst might have other opinions for this

Reply:
Thanks @buyaa-n. 
Note that we are not suppressing for unused lambda/local functions, we don't even analyze them. As soon as there is _some_ reference to the lambda/local function, we will analyze it. This is because we normally analyze these in context of the callers (for explicit invocations) or standalone (for escape cases where lambda/local function can no longer be tracked and may be invoked by other means outside the scope of the analysis). If this scenario is important, it would be easy to enhance the analysis in a follow-up PR, but I felt that is not an important scenario as users will likely not have unused lambda/local functions - compiler also generates a warning for it I believe.

Reply:
Users would also get diagnostics for unused lambda/local function:
![image](https://user-images.githubusercontent.com/10605811/91921092-85eb1800-ec7f-11ea-89a4-71be386d45c3.png)

Reply:
Not analyzing unused local functions feels odd to me. Whether something is used or not usually doesn't remove the requirement for it to be correct. For example, the compiler still binds the body, requires definitive assignment rules etc. I don't see why analyzers would be any different from that.

Reply:
Hmm analogy with binding seems a good case that users would expect diagnostics while they code up a local function. Let me fix this up by tracking this information during flow analysis.

Reply:
@buyaa-n @terrajobst this should be fixed now. Buyaa, the latest commit https://github.com/dotnet/roslyn-analyzers/pull/4113/commits/b4bd3f4912373e2f06b23d2530384b3dd19c67be also fixes couple other analyzer bugs I found while adding tests, please review the analyzer changes. Thanks!

Reply:
Thank you @mavasani that is great